bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/ftp/FtpDirectory.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.ftp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.commons.net.ftp.FTPReply;
+import org.openhab.binding.hapero.internal.handler.HaperoBridgeHandler;
+
+import com.github.drapostolos.rdp4j.spi.FileElement;
+import com.github.drapostolos.rdp4j.spi.PolledDirectory;
+
+/**
+ * {@link HaperoBridgeHandler} is a helper class for FTOP File monitoring
+ * and also implements reading the file contents from an FTP Server.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+public class FtpDirectory implements PolledDirectory {
```

Comment:
I can not use NonNullByDefault here because ovveriding the set function does not work then. Error is: The return type is incompatible with 'Set<FileElement>' returned from PolledDirectory.listFiles() (mismatching null constraints)